test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/BaseE2E.cs

```diff
@@ -0,0 +1,121 @@
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.apps.type;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.command;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.command.queue;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.entity;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.queue;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.queue.operation;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Tests.Invoke;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Tests.Invoke.request;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Tests.Invoke.request.http;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Tests.Invoke.request.queue;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Tests.Invoke.Type;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Util;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.TestLogger;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests
+{
+    /* This class acts as the base class for all the language test case classes.
+     * Takes care of Initial Orchestration and actual flow of the test.
+    */
+    public class BaseE2E
+    {
+        private KafkaE2EFixture kafkaE2EFixture;
+        private Language language;
+        private BrokerType brokerType;
+        private E2ETestInvoker invoker;
+        ITestOutputHelper output;
+        private readonly ILogger logger = TestLogger.TestLogger.GetTestLogger();
+
+        protected BaseE2E(KafkaE2EFixture kafkaE2EFixture, Language language, BrokerType brokerType, ITestOutputHelper output)
+        {
+            this.kafkaE2EFixture = kafkaE2EFixture;
+            
+            this.language = language;
+            this.kafkaE2EFixture.SetLanguage(language);
+            
+            this.brokerType = brokerType;
+            this.kafkaE2EFixture.SetBrokerType(brokerType);
+            
+            this.invoker = new E2ETestInvoker();
+
+            this.kafkaE2EFixture.OrchestrateInitialization();
+            
+            this.output = output;
+        }
+
+        public async Task Test(AppType appType, InvokeType invokeType, HttpRequestEntity httpRequestEntity,
+            KafkaEntity queueEntity, List<string> expectedOutput)
+        {
+            //Send invocation Http request to the function apps 
+            await InvokeE2ETest(appType, invokeType, httpRequestEntity, queueEntity);
+            
+            // wait for the function completion
+            await Task.Delay(60000);
```

Comment:
Can we add a change like [this one](https://github.com/Azure/azure-functions-sql-extension/blob/94dced8cbf10618ef88b68598344749ea93744a8/test/Integration/IntegrationTestBase.cs#L198-L210)? That way, we would be able to move ahead as soon as the Functions host gets started.

Reply:
Note that the check `e.Data.Contains($"http://localhost:{this.Port}/api"))` will only work if there are one or more HTTP triggers registered. I had to update it to `if (e?.Data.Contains("Host lock lease acquired by instance ID") == true) {}` in my case. Also, there are a couple of other issues in the linked code. It doesn't report error if host start times out, and it does not unregister the event handler after the wait is over.